libraries/Networking/src/ESP_Network_Interface.cpp

```diff
@@ -0,0 +1,526 @@
+        log_e("DHCP could not be stopped! Error: %d", err);
+        return false;
+    }
+
+    // Set IPv4, Netmask, Gateway
+    err = esp_netif_set_ip_info(_esp_netif, &info);
+    if(err != ERR_OK){
+        log_e("ETH IP could not be configured! Error: %d", err);
+        return false;
+    }
+    
+    // Set DNS Servers
+    esp_netif_set_dns_info(_esp_netif, ESP_NETIF_DNS_MAIN, &d1);
+    esp_netif_set_dns_info(_esp_netif, ESP_NETIF_DNS_BACKUP, &d2);
+    esp_netif_set_dns_info(_esp_netif, ESP_NETIF_DNS_FALLBACK, &d3);
+
+    // Start DHCPC if static IP was set
+    if(info.ip.addr == 0){
+        err = esp_netif_dhcpc_start(_esp_netif);
+        if(err != ESP_OK && err != ESP_ERR_ESP_NETIF_DHCP_ALREADY_STARTED){
+            log_w("DHCP could not be started! Error: %d", err);
+            return false;
+        }
+    } else {
+        _has_ip = true;
+    }
+
+    return true;
+}
+
+const char * ESP_Network_Interface::getHostname()
+{
+    if(_esp_netif == NULL){
+        return "";
+    }
+    const char * hostname;
+    if(esp_netif_get_hostname(_esp_netif, &hostname)){
+        return NULL;
+    }
+    return hostname;
+}
+
+bool ESP_Network_Interface::setHostname(const char * hostname)
+{
+    if(_esp_netif == NULL){
+        return false;
+    }
+    return esp_netif_set_hostname(_esp_netif, hostname) == 0;
+}
+
+bool ESP_Network_Interface::linkUp()
+{
+    if(_esp_netif == NULL){
+        return false;
+    }
+    return esp_netif_is_netif_up(_esp_netif);
+}
+
+const char * ESP_Network_Interface::ifkey(void)
+{
+    if(_esp_netif == NULL){
+        return "";
+    }
+    return esp_netif_get_ifkey(_esp_netif);
+}
+
+const char * ESP_Network_Interface::desc(void)
+{
+    if(_esp_netif == NULL){
+        return "";
+    }
+    return esp_netif_get_desc(_esp_netif);
+}
+
+String ESP_Network_Interface::impl_name(void)
+{
+    if(_esp_netif == NULL){
+        return String("");
+    }
+    char netif_name[8];
+    esp_err_t err = esp_netif_get_netif_impl_name(_esp_netif, netif_name);
+    if(err != ESP_OK){
+        log_e("Failed to get netif impl_name: %d", err);
+        return String("");
+    }
+    return String(netif_name);
+}
+
+uint8_t * ESP_Network_Interface::macAddress(uint8_t* mac)
+{
+    if(!mac || _esp_netif == NULL){
+        return NULL;
+    }
+    esp_err_t err = esp_netif_get_mac(_esp_netif, mac);
+    if(err != ESP_OK){
+        log_e("Failed to get netif mac: %d", err);
+        return NULL;
+    }
+    // getMac(mac);
+    return mac;
+}
+
+String ESP_Network_Interface::macAddress(void)
+{
+    uint8_t mac[6] = {0,0,0,0,0,0};
+    char macStr[18] = { 0 };
+    macAddress(mac);
+    sprintf(macStr, "%02X:%02X:%02X:%02X:%02X:%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+    return String(macStr);
+}
+
+IPAddress ESP_Network_Interface::localIP()
+{
+    if(_esp_netif == NULL){
+        return IPAddress();
+    }
+    esp_netif_ip_info_t ip;
+    if(esp_netif_get_ip_info(_esp_netif, &ip)){
```

Comment:
This is the local (as opposed to remote) IP address ; it will usually be a global scope address (but sometimes not). Please align the names of IPv4 and IPv6.